src/replication/DealReplicationWorker.ts

```diff
@@ -415,7 +415,7 @@ export default class DealReplicationWorker extends BaseService {
           }
 
           const startDelay = replicationRequest.startDelay ? replicationRequest.startDelay : 20160;
-          const currentHeight = HeightFromCurrentTime();
+          const currentHeight = await this.currentBlockHeight();
```

Comment:
Need to consider the implication on performance. This is called on every deal made. If calling on a 3rd party hosted full node api, they could post a rate limit and fail this.

Recommend: consider moving this outside of the for loop (also need to consider the whole for-loop running time).

Reply:
It is one of the reasons I didn't support devnet till now. There's no good way to get the current height on devnet without performance penalty. Maybe we only do the lotus status on devnet only. Mainnet still use fixed calculation.

Reply:
edited pr